text/2978-stack_based_vec.md

```diff
@@ -86,22 +86,26 @@ Of course, fixed buffers lead to some inflexibility because unlike `Vec`, the un
 # Reference-level explanation
 [reference-level-explanation]: #reference-level-explanation
 
-`ArrayVec` is a contiguous memory block where elements can be collected, threfore, a collection by definition and even though `core::collections` doesn't exist, it is the most natural module placement.
+`ArrayVec` is a contiguous memory block where elements can be collected, therefore, a collection by definition and even though `core::collections` doesn't exist, it is the most natural module placement.
 
-The API basically mimics most of the current `Vec` surface with some tweaks and additional methods to manage capacity. Notably, these additional methods are fallible versions of some well-known functions like `insert` that will return `Result` instead of panicking at run-time.
+The API basically mimics most of the current `Vec` surface with some tweaks to manage capacity. Notably, these additional methods are fallible versions of some well-known functions like `push` that will return `Result` instead of panicking at run-time.
 
 ```rust
 // core::collections
 
 pub struct ArrayVec<T, const N: usize> {
     data: MaybeUninit<[T; N]>,
-    len: usize
+    len: usize,
 }
 
 impl<T, const N: usize> ArrayVec<T, N> {
     // Constructors
 
-    pub unsafe fn from_raw_parts(_ptr: *mut T, _len: usize) -> Self;
+    #[inline]
+    pub fn from_array(array: [T; N]) -> Self;
+
+    #[inline]
+    pub fn from_array_and_len(array: [T; N], len: usize) -> Self;
```

Comment:
```suggestion
    pub fn from_array_and_len(array: [MaybeUninit<T>; N], len: usize) -> Self;
```
I am not sure if that method is valuable, as all cases where ` [MaybeUninit<T>; N]` and a `len`  is used we should "just" use `ArrayVec` directly

Reply:
Beyond the semantic value, storage re-utilization and a possible future function "constification", `from_array` and `from_array_and_len` are right now being used to write unit-tests :)